Cannot compile luajit due to `-std=c11`

```

gcc -MMD -MT build/linux-x64/release/obj/lj_str_hash.c/lj_str_hash.obj -MF build/linux-x64/release/obj/lj_str_hash.c/lj_str_hash.obj.d -std=c11 -O2 -Wall -fvisibility=hidden -I3rd/lua/luajit/src -DLUAJIT_UNWIND_EXTERNAL -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DLUA_MULTILIB="lib" -DLUAJIT_ENABLE_LUA52COMPAT -DLUAJIT_NUMMODE=2 -DNDEBUG -fno-stack-protector -U_FORTIFY_SOURCE -fPIC -msse4.2 -o build/linux-x64/release/obj/lj_str_hash.c/lj_str_hash.obj -c 3rd/lua/luajit/src/lj_str_hash.c
3rd/lua/luajit/src/lj_str_hash.c: In function ‘str_hash_init_random’:
3rd/lua/luajit/src/lj_str_hash.c:188:3: error: implicit declaration of function ‘srandom’; did you mean ‘srand’? [-Wimplicit-function-declaration]
  188 |   srandom(seed);
      |   ^~~~~~~
      |   srand
3rd/lua/luajit/src/lj_str_hash.c:197:24: error: implicit declaration of function ‘random’; did you mean ‘rand’? [-Wimplicit-function-declaration]
  197 |     random_pos[i][0] = random() & POW2_MASK(i+1);
      |                        ^~~~~~
      |                        rand
```

--- a/compile/luajit/make.lua
+++ b/compile/luajit/make.lua
@@ -118,7 +118,8 @@ lm:source_set("lj_str_hash.c") {
         _LARGEFILE_SOURCE,
         LUA_MULTILIB,
         LUAJIT_ENABLE_LUA52COMPAT,
-        LUAJIT_NUMMODE
+        LUAJIT_NUMMODE,
+        "_XOPEN_SOURCE=600"
     },
     flags = lj_str_hash_flags
 }

